.github/workflows/ci-prover-e2e.yml

```diff
@@ -0,0 +1,152 @@
+name: Workflow for testing prover component end-to-end
+on:
+  workflow_call:
+
+jobs:
+  e2e-test:
+    runs-on: [ matterlabs-ci-gpu-l4-runner-prover-tests ]
+    env:
+      RUNNER_COMPOSE_FILE: "docker-compose-gpu-runner-cuda-12-0.yml"
+
+    steps:
+      - uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4
+        with:
+          submodules: "recursive"
+          fetch-depth: 0
+
+      - name: Setup environment
+        run: |
+          echo ZKSYNC_HOME=$(pwd) >> $GITHUB_ENV
+          echo $(pwd)/bin >> $GITHUB_PATH
+          echo IN_DOCKER=1 >> .env
+          echo "SCCACHE_GCS_BUCKET=matterlabs-infra-sccache-storage" >> .env
+          echo "SCCACHE_GCS_SERVICE_ACCOUNT=gha-ci-runners@matterlabs-infra.iam.gserviceaccount.com" >> .env
+          echo "SCCACHE_GCS_RW_MODE=READ_WRITE" >> .env
+          echo "RUSTC_WRAPPER=sccache" >> .env
+          
+          mkdir -p prover_logs
+
+      - name: Start services
+        run: |
+          run_retried docker-compose -f ${RUNNER_COMPOSE_FILE} pull
+          mkdir -p ./volumes/postgres ./volumes/reth/data
+          docker-compose -f ${RUNNER_COMPOSE_FILE} --profile runner up -d --wait
+          ci_run sccache --start-server
+
+      - name: Init
+        run: |
+          ci_run git config --global --add safe.directory /usr/src/zksync
```

Comment:
I think just one line
```
ci_run git config --global --add safe.directory /usr/src/zksync/*
```
would be better.

Reply:
doesn't seem to work...
```
fatal: detected dubious ownership in repository at '/usr/src/zksync'
To add an exception for this directory, call:

	git config --global --add safe.directory /usr/src/zksync
```

Reply:
I'm generally not sure why I need to put all these dirs there, seem to work fine in other CIs, but this one is failing for some reason

Reply:
I don't think I understand how chmod is related here